src/selectors/visibleBreakpoints.js

```diff
@@ -47,30 +48,33 @@ function isVisible(breakpoint, selectedSource) {
 /*
  * Finds the breakpoints, which appear in the selected source.
  */
-export const getVisibleBreakpoints = createSelector(
+export const getVisibleBreakpoints: Selector<?(Breakpoint[])> = createSelector(
   getSelectedSource,
   getBreakpointsList,
-  (selectedSource: Source, breakpoints: Breakpoint[]) => {
-    if (!selectedSource) {
+  (selectedSource: ?Source, breakpoints: Breakpoint[]) => {
+    if (selectedSource == null) {
       return null;
     }
 
+    // FIXME: Even though selectedSource is checked above, it fails type
+    // checking for isVisible
+    const source: Source = selectedSource;
```

Comment:
:/

Reply:
This is because type refinements in outer functions don't stay refined inside callbacks. I think this is a very reasonable workaround for that and doesn't need a FIXME.